File: backend/src/main/java/app/netbooks/backend/models/PlanEdition.java
package app.netbooks.backend.models;

import java.math.BigDecimal;
import java.sql.Date;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class PlanEdition {
    private Integer id;
    private Integer plan;
    private Integer numSubscribers;
    private BigDecimal price;
    private Date startedIn;
    private Date closedIn;
    private Boolean available;
};

File: frontend/src/models/plan_edition.ts
import { Decimal } from "decimal.js";

export type PlanEditionData = {
  id: number;
  plan: number;
  numSubscribers: number;
  price: number;
  startedIn: number;
  closedIn: number;
};

export class PlanEdition {
  readonly id: number;
  readonly plan: number;
  readonly numSubscribers: number;
  readonly price: Decimal;
  readonly startedIn: Date;
  readonly closedIn?: Date;

  constructor(data: PlanEditionData) {
    this.id = data.id;
    this.plan = data.plan;
    this.numSubscribers = data.numSubscribers;
    this.price = new Decimal(data.price);
    this.startedIn = new Date(data.startedIn);
    if (data.closedIn) this.closedIn = new Date(data.closedIn);
  }
}
